csun-metalab/media
# Task photo upload

# Summary
This branch as described includes the photo upload feature as well as general code clean-up.

# To test
The new feature is the file upload which accepts a file and a secret key.
You'll need to:
+ Place `SECRET_KEY` inside your `.env` and set it to something.
+ Place `IMAGE_UPLOAD_LOCATION` inside your `.env` and give it the path to the project's `public/` folder.
ex: `/home/user/<user_name>/media/public/` - For Mac/Linux/Homestead
`C:\Users\<user_name>\media\public\` or `C:\wamp\www\media\public\` - For Windows using WAMP
+ Using [Postman](https://www.getpostman.com/), [Insomnia](https://insomnia.rest/) or PHPStorm's built-in tool to test RESTful web service, issue a post request to
`<localhost>/api/1.0/<some_string>/post?secret_key=` and include a form input value of profile_image which is mapped to the image file you want to upload.
**Note:** make sure the value of `?secret_key=` matches your `.env` value for `SECRET_KEY=`

Once these requirements are met and you've uploaded a file, you should see a success JSON response along with a new folder inside the public directory that matches the value you gave to `<some_string>`  in the post request, and finally the image file you uploaded with the name `avatar.jpg`.
